### DISCUSSION TITLE
Telegram UI

### DISCUSSION THREAD

**Discussion by hardBSDk** [markqvist/Reticulum#133] [Requests for New Features]
This is a long-term feature request, after Sideband maturity.

It would be wonderful if it use Telegram client UI, as it's very mature and fancy.

(TDLib is required? IDK, Telegram X uses it and will replace official client).

Code suggestions:

https://github.com/DrKLO/Telegram
https://github.com/telegramdesktop/tdesktop

**Comment by markqvist** [comment]
It is an interesting proposal. Probably not something I will be working on for the foreseeable future, but someone else could make another LXMF client with an UI like that.

**Comment by maxbrito500** [comment]
Hello,

I have some experience writing Telegram code before, their bots are a modern replacement for websites. However, that is also the only interaction (AFAIK) that telegram is providing. So you'd have to run a bot at some server to get basic interaction.

From there, the bot is the bridge between your telegram account and your contacts. The bot could likely receive/send messages on your behalf and show them as telegram notifications. There isn't yet a way to create ghost users inside a bot channel, but I suppose you can quote reply to a specific text and from there the bot knows who to dispatch your answer.

All of this is possible, albeit I'm not understanding the usage scenario. Telegram needs Internet, you need a server to run the bot and reticulum is mostly intended to keep some distance from Internet. At some point why would you even need reticulum for everyday internet-connected things?

**Reply by RFnexus** [reply]
Not directly related to using Telegram's frontend as a UI for Sideband, but there is a functional LXMF to Telegram bridge here:
https://github.com/SebastianObi/LXMF-Tools
https://github.com/SebastianObi/LXMF-Tools/tree/main/lxmf_bridge_telegram

**Comment by markqvist** [comment]
Also, anyone interested in an LXMF client with a more "web-ish" UI should have a look at the [Reticulum Webchat](https://github.com/liamcottle/reticulum-webchat) project that @liamcottle is working on, and has already made great progress on.
